feature/authorization-registration-forms

FellonOne opened the pull request:
- add authorization and registration forms;

Commit:
add authorization and registration pages;
add Formik library;

Commit:
add authorization and registration pages;
add Formik library;

Commit:
add authorization and registration pages;
add Formik library;

Commit:
add authorization and registration pages;
add Formik library;

art4vm commented on app/src/assets/scss/_authentication.scss:
```diff
@@ -0,0 +1,19 @@
+.authentication {
+    width: auto !important;
+    min-height: auto !important;
```
лучше избегать таких переопределений

kdeit replied:
Где-то прочитал правило (и взял его за стандарт), что использование !important = явный фэйл.
Особо критичные товарищи говорят, что если ты что-то где-то переопределяешь (не только через important), то это также фейл. Но это через чур радикально.

art4vm commented on app/src/assets/scss/_authentication.scss:
```diff
@@ -0,0 +1,19 @@
+        line-height: 50px;
+        margin-top: 20px;
+    }
+}
```
Параметры цвета в css наследуются. Зачем их определять в каждом элементе?
Желательно вынести в переменные цвета и размера шрифта
Предлагаю размер шрифта указывать в em
Для отступов есть миксины и классы

Возможно стоит это делать отдельным патчем

art4vm commented on app/src/components/FormControl/FormControl.scss:
```diff
@@ -0,0 +1,43 @@
+        color: indianred;
+        margin-top: 5px;
+    }
+}
```
Про цвета, отступы и шрифты см предыдущий коммент

art4vm commented on app/src/components/FormControl/FormControl.tsx:
```diff
@@ -0,0 +1,42 @@
+                </div>
+            </div>
+        );
+    });
```
Больше похоже, что стиль input-container сам является блоком

art4vm commented on app/src/components/FormControl/FormControl.tsx:
```diff
@@ -0,0 +1,42 @@
+    const initialValues: FormControlInputFields = Object.keys(fields).reduce((res, key) => {
+        res[key] = '';
+        return res;
+    }, {});
```
поля не всегда могуть быть пустыми изначально, например при изменении параметров пользователя

art4vm commented on app/src/pages/Authorization/Authorization.tsx:
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import * as Yup from 'yup';
+import {Button} from '../../components/Button';
```
сам же вроде предлагал использовать index.ts в /components  и импортировать из него

FellonOne replied:
Увидел в каком-то патче, что удалили index.ts у Button
Подумал, что отказались и не стал добавлять
Мы все еще его используем? если да, то добавлю

art4vm replied:
Я удалил, тк он перестал быть нужен, но Index который в корне каталога components остался

art4vm commented on package.json:
```diff
@@ -38,9 +38,11 @@
   },
   "dependencies": {
     "express": "4.17.1",
+    "formik": "^2.2.5",
```
Тут и ниже нужно залочить версию

art4vm replied:
А чего сам formik не залочил 😃

art4vm commented on app/src/pages/Authorization/Authorization.scss:
```diff
@@ -0,0 +1,5 @@
+    min-width: 180px;
+    font-size: 16px;
+    margin-left: 60px;
+}
```
См предыдущие комменты по стилям

Commit:
update form control component;

Commit:
update form control component;

art4vm commented on app/src/pages/Authorization/Authorization.tsx:
```diff
@@ -0,0 +1,54 @@
+                    }}
+                    onSubmit={(formData) => {
+                        // eslint-disable-next-line no-console
+                        console.log(formData);
```
нужные действия уже есть, тч думаю стоит сразу впилить сюда вызов авторизации

FellonOne replied:
Думал добавить привязку к сагам следующим пулл-реквестом
Там как раз есть задачи на сервис Авторизации/Регистрации

+ в след пулл реквесте связать данные страницы друг с другом (добавить роутинг регистрация <=> авторизация) и тд

art4vm replied:
тут к сагам не нужно ничего привязывать, нужно подключить redux через react-redux и вызвать action в который в виде **объекта** передать поля.

art4vm commented on app/src/pages/Registration/Registration.tsx:
```diff
@@ -0,0 +1,75 @@
+                            title: 'Create new password: '
+                        }
+                    }}
+                    onSubmit={(formData) => {
```
так же как и с авторизацией. Только нужно добавить обработку в sagas.ts

art4vm replied:
обработчик в sagas уже добавил

jarith replied:
Предлагаю отдельным PR

art4vm commented on app/src/components/FormControl/FormControl.tsx:
```diff
@@ -0,0 +1,39 @@
+    touched: FormikTouched<FormControlInputFields>
+) =>
+    Object.entries(fields).map(([fieldName, {placeholder, type, title}]) => (
+        <div className="form-block__input-container" key={fieldName} style={{color: 'white'}}>
```
input-container выглядит как отдельный контейнер, тем более инпут может быть вне формы

art4vm commented on app/src/pages/Registration/Registration.tsx:
```diff
@@ -0,0 +1,75 @@
+import {FormControl} from '../../components/FormControl/FormControl';
+import './Registration.scss';
+
+const RegistrationSchema = Yup.object().shape({
+    login: Yup.string()
+        .min(5, 'min length 5 symbols')
+        .max(12, 'max length 12 symbols')
+        .required('field must be required'),
+    password: Yup.string()
+        .min(4, 'min length 4 symbols')
+        .max(16, 'max length 16 symbols')
+        .required('field must be required'),
+    phoneNumber: Yup.string()
+        .min(8, 'min length 8 symbols')
+        .max(16, 'max length 16 symbols')
+        .required('field must be required'),
+    firstName: Yup.string(),
+    secondName: Yup.string()
+});
```
Нужны валидаторы на формат поля

FellonOne replied:
Имелось ввиду на формат телефона?

art4vm commented on app/src/components/FormControl/FormControl.tsx:
```diff
@@ -0,0 +1,39 @@
+                <div className="form-block__title">{title}</div>
+                <Field placeholder={placeholder} className="form-block__input" type={type} name={fieldName} />
+            </label>
+            <div className="form-block__validation-message">
+                {touched[fieldName] && errors[fieldName] ? errors[fieldName] : ''}
+            </div>
```
У формика есть компонент ErrorMessage, он тут не подходит?

Commit:
add initialValue in FormControl;
update css;
fixed depend. version;
add ErrorMessage by Formik;

Commit:
add initialValue in FormControl;
update css;
fixed depend. version;
add ErrorMessage by Formik;

art4vm commented on app/src/pages/Registration/Registration.tsx:
```diff
@@ -0,0 +1,73 @@
+        .min(4, 'min length 4 symbols')
+        .max(16, 'max length 16 symbols')
+        .required('field must be required'),
+    phoneNumber: Yup.string().phone('*', false, 'phone number is not valid').required('field must be required'),
```
а по какому формату он проверяет телефон?
я имел ввиду проверку всех полей по заданному формату, где просто regexp задаешь. Кроме телефона есть еще и email, да и пароль и логин принято проверять на допустимые символы

FellonOne replied:
Формат строка вида 901234567 90 - код оператора 123 45 67  номер
либо +998 (код страны) + то что выше
либо 998 (код страны) + то что выше

Поле email добавил + проверку 
Насчет логина и тд, там вроде как будут допустим любые символы
можно добавить доп проверку по регулярке, но думаю на данном этапа можно пока оставить так

kdeit commented on app/src/assets/scss/_authentication.scss:
```diff
@@ -0,0 +1,16 @@
+    &__footer {
+        display: flex;
+        justify-content: space-between;
+        font-size: 16px;
```
Посмотри миксины для типографики. Я бы предложил их использовать.
Т.е. в них определить все начертания текста в проекте и всё, что касается текста писать через миксины.
Очень часто после вёрстки дизайнер смотрит и говорит «Неее. Надо, конечно, этот текст поменьше сделать» и что-б по проекту не искать в миксине поменять и всё.
+ наглядно видишь все начертания. А их много лучше не использовать (иначе зоопарк).

kdeit commented on app/src/components/button/Button.tsx:
```diff
@@ -3,14 +3,15 @@ import {Props} from './types';
 Button.defaultProps = {
-    className: 'btn t-main'
+    className: 'btn t-main',
+    submit: false
```
button type еще может быть reset. Так что тут boolean не корректен имхо

kdeit commented on app/src/pages/Registration/Registration.tsx:
```diff
@@ -0,0 +1,73 @@
+    return (
+        <main className="ui">
+            <div className="ui__inner authentication">
+                <h4 className="t-title authentication__title">
```
Если это первый заголовок на странице, то семантически правильно `<h1>` и далее по вложенности.
А то ещё прибежит СЕО-шник и будет бомбить :)

jarith replied:
> Если это первый заголовок

Только не первый, а основной, который описывает содержимое страницы. То есть, если это страница регистрации, то через `<h1>` разумно разметить заголовок "Регистрация" и не важно, первым он идет или последним

jarith commented on app/src/pages/Authorization/Authorization.tsx:
```diff
@@ -0,0 +1,54 @@
+    return (
+        <main className="ui">
+            <div className="ui__inner authentication">
+                <h4 className="t-title authentication__title">
```
`<h1>` больше подойдет, т. к. это главный заголовок страницы

jarith commented on app/src/pages/Authorization/Authorization.tsx:
```diff
@@ -0,0 +1,54 @@
+        <main className="ui">
+            <div className="ui__inner authentication">
+                <h4 className="t-title authentication__title">
+                    <span>Authorization</span>
```
```suggestion
                    Authorization
```

jarith commented on app/src/pages/Authorization/Authorization.tsx:
```diff
@@ -0,0 +1,54 @@
+                            placeholder: 'input your password',
+                            title: 'Your password: '
+                        }
+                    }}
```
Предлагаю не инлайнить этот объект, а объявить отдельно перед компонентом

jarith commented on app/src/pages/Authorization/Authorization.tsx:
```diff
@@ -0,0 +1,54 @@
+                    }}
+                >
+                    <footer className="authentication__footer">
+                        <div>Registration</div>
```
Тут же должна быть ссылка на страницу регистрации? Если так, то ссылкой и нужно сделать

jarith commented on app/src/pages/Registration/Registration.tsx:
```diff
@@ -0,0 +1,73 @@
+                            placeholder: 'Input your password',
+                            title: 'Create new password: '
+                        }
+                    }}
```
Тут прям точно стоит вытащить объект из компонента

jarith commented on app/src/pages/Registration/Registration.tsx:
```diff
@@ -0,0 +1,73 @@
+                    }}
+                >
+                    <footer className="authentication__footer">
+                        <div>Authorization</div>
```
Соответственно, тут надо бы ссылку на страницу авторизации

Commit:
update Registration and Authorization pages;

Commit:
update Registration and Authorization pages;

Commit:
fix all issues; rebase with develop;

Commit:
fix all issues; rebase with develop;

Commit:
fix all issues; rebase with develop;

Commit:
fix all issues; rebase with develop;

art4vm committed:
Добавлена saga для регистрации (#43) (#11)

art4vm committed:
Добавлена saga для регистрации (#43) (#11)

art4vm committed:
Feature/52 refactoring (#13)

* Основной шрифт заменен на 'Press Start 2P' (#55)

* Рефакторинг (#52)

Рефакторинг стилей
  Исправлены названия
  Приведены к camelCase параметры sass
  Нормализованы св-ва согласно правилам stylelint
  Вынесены в параметры URL assets
  Добавлены css св-ва с цветами
  Изменены единицы размеров шрифтов с px на rem

Добавлено выполнение stylelint в pre-commit
Поправлено описание в README

art4vm committed:
Feature/52 refactoring (#13)

* Основной шрифт заменен на 'Press Start 2P' (#55)

* Рефакторинг (#52)

Рефакторинг стилей
  Исправлены названия
  Приведены к camelCase параметры sass
  Нормализованы св-ва согласно правилам stylelint
  Вынесены в параметры URL assets
  Добавлены css св-ва с цветами
  Изменены единицы размеров шрифтов с px на rem

Добавлено выполнение stylelint в pre-commit
Поправлено описание в README

jarith commented:
@FellonOne  Остался коммент про миксины и надо бы конфликты порешать

В остальном все отлично

Commit:
fix issues;

Commit:
fix issues;

Commit:
fix conflicts;

Commit:
fix conflicts;

Commit:
fix conflict, part II;

Commit:
Merge branch 'features/authorization-form' of https://github.com/PraktikumSeoulDevTeam/castleofghosts into features/authorization-form

# Conflicts:
#	app/src/components/index.ts
#	app/src/pages/AuthorizationPage/Authorization.tsx
#	app/src/pages/RegistrationPage/Registration.tsx
